editor.planx.uk/src/pages/FlowEditor/components/Sidebar/Publish/AlteredNodes.tsx

```diff
@@ -0,0 +1,237 @@
+  background: theme.palette.secondary.dark,
+  color: theme.palette.text.primary,
+  borderRadius: theme.shape.borderRadius,
+  border: `1px solid rgba(0,0,0,0.08)`,
+}));
+
+export const AlteredNodeListItem = (props: { node: AlteredNode }) => {
+  const { node } = props;
+  let text, data;
+
+  if (node.id === "_root") {
+    text = "Changed _root service by adding, deleting or re-ordering nodes";
+  } else if (node.id === "0") {
+    text = `The entire _root service will be published for the first time`;
+  } else if (node.id && Object.keys(node).length === 1) {
+    text = `Deleted node ${node.id}`;
+  } else if (node.type && node.data) {
+    text = `Added/edited ${TYPES[node.type]}`;
+    data = JSON.stringify(node.data, null, 2);
+  } else {
+    text = `Added/edited ${TYPES[node.type]}`;
+  }
+
+  return (
+    <ListItem key={node.id} disablePadding sx={{ display: "list-item" }}>
+      <Typography variant="body2">{text}</Typography>
+      {data && (
+        <pre
+          style={{
+            fontSize: ".8em",
+            whiteSpace: "pre-line",
+            overflowWrap: "break-word",
+            wordWrap: "break-word",
+          }}
+        >
+          {data}
+        </pre>
+      )}
+    </ListItem>
+  );
+};
+
+interface AlteredNodesSummary {
+  title: string;
+  portals: ExternalPortal[];
+  updated: number;
+  deleted: number;
+}
+
+export const AlteredNodesSummaryContent = (props: {
+  alteredNodes: AlteredNode[];
+  lastPublishedTitle: string;
+  history?: HistoryItem[];
+}) => {
+  const { alteredNodes, lastPublishedTitle, history } = props;
+  const [expanded, setExpanded] = useState(false);
+  const comments = history?.filter((item) => item.type === "comment") || [];
+  const operations = history?.filter((item) => item.type === "operation") || [];
+
+  const isPlatformAdmin = useStore.getState().user?.isPlatformAdmin;
+
+  const changeSummary: AlteredNodesSummary = {
+    title: lastPublishedTitle,
+    portals: [],
+    updated: 0,
+    deleted: 0,
+  };
+
+  alteredNodes.map((node) => {
+    if (node.id === "0") {
+      changeSummary["title"] =
+        "You are publishing the main service for the first time.";
+    } else if (node.id && Object.keys(node).length === 1) {
+      changeSummary["deleted"] += 1;
+    } else if (node.type === TYPES.InternalPortal) {
+      if (node.data?.text?.includes("/") && !node.data?.text?.includes(" ")) {
+        changeSummary["portals"].push({ ...node.data, flowId: node.id });
+      }
+    } else if (node.type) {
+      changeSummary["updated"] += 1;
+    }
+
+    return changeSummary;
+  });
+
+  return (
+    <Box>
+      <Typography variant="h3" component="h2" gutterBottom pt={1}>
+        {`Changes to your service since last publish`}
+      </Typography>
+      {comments.length > 0 && (
+        <List sx={{ listStyleType: "none" }}>
+          {comments.map((comment) => (
+            <ListItem key={comment.id} disablePadding>
+              <ListItemText
```

Comment:
This is leading to `validateDomNesting` errors - it looks like it's a `p` or `span` wrapping other elements.

We can't use a `component="div"` prop on this element - is there an alternative way of rendering this?

<img width="608" alt="image" src="https://github.com/user-attachments/assets/b10e6d53-8a3e-4926-a50f-94d8db3ff6e1" />